File: models/lab.js
var mongoose = require('mongoose')
var Schema = mongoose.Schema;

var labSchema = new Schema({
    name:{
        type : String,
    },
    image:{
        type : String,
    },
    location:{
        type : String,
    },
    email: {
        type: String,
        require: true
    },
    password: {
        type: String,
        require: true
    },
    phone: {
        type: String,
    },
    longitude: {
        type: Number,
    },
    latitude: {
        type: Number,
    },
    homeVisit: {
        type: Boolean,
    },
    test: {
        type: Array,
    },
    rating: {
        type: Number,
    },
    openTime:{
        type: String,
    },
    closeTime: {
        type: String
    },
    ownerName: {
        type: String
    },
    about:{
        type: String
    }

})
module.exports = mongoose.model('Lab',labSchema)

File: flutter_application/lib/models/lab.dart
import 'package:objectid/objectid.dart';

class lab {
  String id = "";
  String name = "";
  String email = "";
  String openTime = "";
  double latitude = 0.0;
  double longitude = 0.0;
  int rating = 0;
  String location = "";
  String ownerName = "";
  String about = "";
  List? test;
  bool homeVisit = false;
  String closeTime = "";
  String phone = "";
  List? unavailable;

  lab(
      this.id,
      this.name,
      this.email,
      this.latitude,
      this.longitude,
      this.rating,
      this.test,
      this.unavailable,
      this.openTime,
      this.ownerName,
      this.closeTime,
      this.about,
      this.homeVisit);
  lab.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    name = json['name'];
    phone = json['phone'];
    about = json['about'];
    latitude = json['latitude'];
    longitude = json['longitude'];
    location = json['location'];
    ownerName = json['ownerName'];
    openTime = json['openTime'];
    closeTime = json['closeTime'];
    email = json['email'];
    test = json['test'].cast<String>();
    unavailable = json['unavailableTime'].cast<String>();
  }
}
